athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchDomainMapProvider.java

```diff
@@ -127,6 +127,15 @@ protected ElasticsearchDomainMapProvider(boolean autoDiscoverEndpoint)
         }
     }
 
+    /**
+     * Gets an Amazon Elasticsearch client.
+     * @return Amazon Elasticsearch client.
+     */
+    protected AWSElasticsearch getElasticsearchClient()
+    {
+        return AWSElasticsearchClientBuilder.defaultClient();
+    }
```

Comment:
Can we inject a factory instead? This way the test can inject a mock instead of using a spy / partial mock?

Reply:
Done.